### PR TITLE
[chip, dv] Fix SVA errors due to X

### PR DESCRIPTION
**Pull request by weicaiyang**
Fixed 2 issues in chip-level sim
1. In foundry sim, we have different delays for combination pathes. Then, we see CSB and CIO take different time to reach spi_host from pads. We used core clk to check CIO is known, which isn't the right clock. Switched to use SPI clock.

2. In foundry sim, due to the pullup delay, when no one drives SPI CIO, the pullup may takes some delay to take effect and assign weak high to CIO, which leads to unknown value. However, this unknown value won't be used by either side. Removed checking `passthrough_o.s` is known, as it's an input passed through to spi device. Changed the SVA to do a connectivity check. Thanks @msfschaffner for helping to look into this.

This will fix the foundry CI failure.

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c [hunk 1]
```diff
@@ -205,8 +205,8 @@ bool test_main(void) {
 
   // Turn off the AON timer hardware completely before exiting.
   aon_timer_testutils_shutdown(&aon_timer);
-  if (kDeviceType == kDeviceSimDV) {
-    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
-  }
+  // if (kDeviceType == kDeviceSimDV) {
```
Comment (a-will) [inline sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c:208]: Is this change related? Or should it be dropped?

**Review by a-will — APPROVED** [review]
Thanks for fixing this!

File: sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c (hunk 1 above)
Comment (msfschaffner) [reply sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c:208]: This is actually an interesting bug that we need to look into.
It is present in another test also.
See #16597.

But same question from my side - is this hunk related to this PR?

**Review by msfschaffner — APPROVED** [review]

File: sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c (hunk 1 above)
Comment (weicaiyang) [reply sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c:208]: oh no, I was debugging 2 issues in a same branch and accidentally got this in. Let me remove this one.
